homeassistant/components/rainbird/__init__.py

```diff
@@ -32,17 +33,31 @@
     SENSOR_TYPE_RAINDELAY: ["Raindelay", None, "mdi:water-off"],
 }
 
+TRIGGER_TIME_SCHEMA = vol.All(cv.time_period, cv.positive_timedelta)
+
+
+def _validator(config):
+    config = copy.deepcopy(config)
+    for zone, value in config[CONF_ZONES]:
+        if not value[CONF_TRIGGER_TIME]:
+            value[CONF_TRIGGER_TIME] = config[CONF_TRIGGER_TIME]
+    return config
+
+
 ZONE_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
-        vol.Optional(CONF_TRIGGER_TIME): cv.positive_int,
-    }
+    vol.All(
+        {
+            vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+            vol.Optional(CONF_TRIGGER_TIME): TRIGGER_TIME_SCHEMA,
+        },
+        _validator,
```

Comment:
The validator should be applied in the `CONTROLLER_SCHEMA` after the dict with all items.